test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/AzureFunctions/AzureFunctionsServiceTests.cs

```diff
@@ -223,13 +218,39 @@ public void GetAzureFunctionsNet5()
 
             GetAzureFunctionsParams parameters = new GetAzureFunctionsParams
             {
-                filePath = testFile
+                FilePath = testFile
             };
 
             GetAzureFunctionsOperation operation = new GetAzureFunctionsOperation(parameters);
 
             Exception ex = Assert.Throws<Exception>(() => { operation.GetAzureFunctions(); });
-            Assert.AreEqual(SR.SqlBindingsNet5NotSupported, ex.Message);
+            Assert.That(ex.Message, Is.EqualTo(SR.SqlBindingsNet5NotSupported));
+        }
+
+        /// <summary>
+        /// Verify getting the routes of Azure Functions in a file
+        /// </summary>
+        [Test]
+        public void GetAzureFunctionsRoute()
+        {
+            string testFile = Path.Join(testAzureFunctionsFolder, "AzureFunctionsRoute.cs");
+
+            GetAzureFunctionsParams parameters = new GetAzureFunctionsParams
+            {
+                FilePath = testFile
+            };
+
+            GetAzureFunctionsOperation operation = new GetAzureFunctionsOperation(parameters);
+            GetAzureFunctionsResult result = operation.GetAzureFunctions();
+
+            Assert.That(result.AzureFunctions.Length, Is.EqualTo(7));
+            Assert.That(result.AzureFunctions[0].Route, Is.EqualTo("withRoute"));
+            Assert.That(result.AzureFunctions[1].Route, Is.EqualTo("{interpolated}String"));
+            Assert.That(result.AzureFunctions[2].Route, Is.EqualTo("$withDollarSigns$"));
+            Assert.That(result.AzureFunctions[3].Route, Is.EqualTo("withRouteNoSpaces"));
+            Assert.That(result.AzureFunctions[4].Route, Is.EqualTo("withRouteExtraSpaces"));
+            Assert.That(result.AzureFunctions[5].Route, Is.Null, "Route specified as null should be null");
+            Assert.That(result.AzureFunctions[6].Route, Is.Null, "No route specified should be null");
```

Comment:
how about another route with an empty string? is that valid?

Reply:
What do you mean "another" route? Or you just mean add another one to test for `Route = ""`?

Reply:
Right, `Route = ""`